File: src/core/schemas/storyboard.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


class NarrationBeat(BaseModel):
    id: int
    text: str


class StoryboardScene(BaseModel):
    id: int
    narration_segment: str
    scene_description: str
    reasoning: str
    keywords: list[str]


class StoryJudgeResult(BaseModel):
    narrative_quality: float
    brief_adherence: float
    context_adherence: float
    total_score: float
    feedback: str
    decision: Literal["approve", "revise"]


class JudgeResult(BaseModel):
    score: float
    feedback: str
    decision: Literal["approve", "revise", "escalate"]


class StoryboardOutput(BaseModel):
    user_brief: str
    story: str
    narration_beats: list[NarrationBeat]
    scenes: list[StoryboardScene]
    story_judge_result: StoryJudgeResult | None = None
    story_revision_count: int = 0
    judge_result: JudgeResult
    revision_count: int

File: frontend/src/types/storyboard.ts
export interface NarrationBeat {
  id: number
  text: string
}

export interface StoryboardScene {
  id: number
  narration_segment: string
  scene_description: string
  reasoning: string
  keywords: string[]
}

export interface StoryJudgeResult {
  narrative_quality: number
  brief_adherence: number
  context_adherence: number
  total_score: number
  feedback: string
  decision: string
}

export interface JudgeResult {
  score: number
  feedback: string
  decision: string
}

export interface StoryboardOutput {
  user_brief: string
  story: string
  narration_beats: NarrationBeat[]
  scenes: StoryboardScene[]
  story_judge_result: StoryJudgeResult | null
  story_revision_count: number
  judge_result: JudgeResult
  revision_count: number
}
